File: week1-c/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int findCoins(int cents, int coin_array[], int position);

int main(void)
{
    bool error = true;
    float dollars = 0.00;
    int coin_array[4] = {25, 10, 5, 1};
    int array_size = (int)(sizeof coin_array / sizeof coin_array[0]);
    do
    {
        //ask for the changed owed
        dollars = get_float("Change owed: ");
        if (dollars >= 0)
        {
            error = false;
        }
    }
    while (error);

    //round input to cents
    int cents = round(dollars * 100);
    int coins = 0;
    //recursive function call to get coins ammount
    coins = findCoins(cents, coin_array, 0);
    printf("%i\n", coins);
}
int findCoins(int cents, int coin_array[], int position)
{
    //get number of coins of selected one
    int coins = cents / coin_array[position];
    int remain = cents % coin_array[position];
    //if there's a remainder we need to go to the next one
    if (remain != 0)
    {
        coins += findCoins(remain, coin_array, position + 1);
    }
    return coins;
}

File: pset06/cash/cash.py
#!/bin/python3
import cs50
import math

def main():

    # array of coins
    coin_array = [25, 10, 5, 1]

    while True:
        # ask for the change owed
        dollars = cs50.get_float("Change owed: ")
        if dollars >= 0:
            break
    
    # round input to cents 
    cents = round(dollars * 100)
    coins = findCoins(cents, coin_array, 0);


    # recursive function call to get coins amount
    print("", coins);


# recursive function
def findCoins(cents, coin_array, position):
    # get the number of coins of selected
    coins = round (cents // coin_array[position])
    remain = round(cents % coin_array[position])
    
    # this is the condition to exit from recursion
    if remain != 0:
        # call function recursively
        coins += findCoins(remain, coin_array, position +1)
    return coins


main()